modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/common/main/infrastructure/OffsetDateTime.kt.mustache

```diff
@@ -0,0 +1,89 @@
+package {{packageName}}.infrastructure
+
+import kotlinx.datetime.*
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+/**
+ * Represents a [LocalDateTime] and the respective [TimeZone] of it.
+ */
+@Serializable(with = OffsetDateTime.Companion::class)
+public data class OffsetDateTime(val dateTime: LocalDateTime, val timeZone: TimeZone) {
```

Comment:
Store a `LocalDateTime` and `ZoneOffset` instead - a `TimeZone` can represent more complex zone id representations such as "Europe/Paris" or "UTC+3". A `ZoneOffset`, on the other hand, is exactly what is represented in the RFC3339 / ISO8601 format. This is in line with [`java.time.OffsetDateTime`](https://docs.oracle.com/javase/8/docs/api/java/time/OffsetDateTime.html), which is likely what `kotlinx.datetime` would use as a reference if implementing a similar class.

In `kotlinx.datetime` the `ZoneOffset` can be retrieved from the `TimeZone` at an `Instant`, so in the case where we have a `LocalDateTime` and `TimeZone`

```kotlin
val zoneOffset = with(timeZone) { offsetAt(localDateTime.toInstant()) }
```

Reply:
`kotlinx.datetime` is just a bit too cutting-edge 🙂 [They have this open issue of parsing & formatting common interchange formats on the roadmap](https://github.com/Kotlin/kotlinx-datetime#knownopen-issues-work-tbd).

Reply:
Yeah you're right

Reply:
So, the problem is that the offset isn't really something you can parse... (no `of()` or any kind of public constructor)

Reply:
Okay changed it, but for now it's a bit ugly until Kotlin/kotlinx-datetime#91 is addressed